packages/common/src/api/tan-query/events/useRemixContest.ts

```diff
@@ -1,38 +1,30 @@
-import { EventEventTypeEnum } from '@audius/sdk'
-import dayjs from 'dayjs'
+import { EventEntityTypeEnum, EventEventTypeEnum } from '@audius/sdk'
 
 import { Event } from '~/models/Event'
 import { ID } from '~/models/Identifiers'
 
 import { SelectableQueryOptions } from '../types'
 
 import { useEventsByEntityId } from './useEventsByEntityId'
-import { EventsByEntityIdOptions } from './utils'
-
-type UseRemixContestOptions<TResult = Event[]> = SelectableQueryOptions<
-  Event[],
-  TResult
-> &
-  EventsByEntityIdOptions
 
 /**
  * Hook to fetch the remix contest event for a given entity ID.
  * Returns the first active RemixContest event found, or null if none exists.
  */
 export const useRemixContest = (
   entityId: ID | null | undefined,
-  options?: UseRemixContestOptions
+  options?: SelectableQueryOptions<Event[], Event>
 ) => {
-  const now = dayjs()
-  const eventsQuery = useEventsByEntityId<Event | null>(entityId, {
-    ...options,
-    select: (events) =>
-      events.find(
-        (event) =>
-          event.eventType === EventEventTypeEnum.RemixContest &&
-          dayjs(event.endDate).isAfter(now)
-      ) ?? null
-  })
+  const eventsQuery = useEventsByEntityId(
+    {
+      entityId,
+      entityType: EventEntityTypeEnum.Track,
+      eventType: EventEventTypeEnum.RemixContest,
+      filterDeleted: true,
+      limit: 1
+    },
+    { ...options, select: (events) => events[0] }
```

Comment:
doesn't this remove the logic that only selects an ongoing event?
or do we want it this way to show UI for "remix contest happened but is now over."

Reply:
yeah good question, if it doesn't already it should sort the "latest" event first would be my hope?